recipes/full_finetune_single_device.py

```diff
@@ -557,15 +533,18 @@ def _setup_data(
             drop_last=True,
         )
 
+        if dataloader_state_dict:
+            dataloader.load_state_dict(dataloader_state_dict)
+
         return dataloader
 
     def save_checkpoint(self, epoch: int) -> None:
         """
         Save state dict to file. The recipe save_checkpoint method is responsible for
         correctly creating the checkpoint dict and passing to the checkpointer.
         """
-        ckpt_dict = {training.MODEL_KEY: self._model.state_dict()}
-        # if training is in-progress, checkpoint the optimizer state as well
+        ckpt_dict: dict[str, Any] = {training.MODEL_KEY: self._model.state_dict()}
```

Comment:
What's with the typing here? Do we need it for some reason?

Reply:
Linter was complaining.

Reply:
But no, not strictly necessary.

Reply:
Oh weird. Btw you wanna open a "good first issue" we should consolidate typehints on the non-typing versions